website/app_test.js

```diff
@@ -146,3 +148,19 @@ test(function formatBytesPatterns() {
   assertEqual(formatBytes(180000000), "171.66 MB");
   assertEqual(formatBytes(18000000000), "16.76 GB");
 });
+
+test(function formatSecondsPatterns() {
+  assertEqual(formatSeconds(10), "0 min");
+  assertEqual(formatSeconds(100), "2 min");
+  assertEqual(formatSeconds(1000), "17 min");
+  assertEqual(formatSeconds(10000), "167 min");
+})
+
+test(async function getTravisDataSuccess() {
+  try {
+    const data = await getTravisData();
```

Comment:
I'm not super happy that this depends on an internet connection to run. Obviously we need an internet connection to get this data - but we don't need an internet connection to test the response parsing. Would it be possible to store a sample of the expected JSON into the repo:
```
curl https://api.travis-ci.com/repos/denoland/deno/builds?event_type=pull_request > tools/testdata/travis_benchmark.json
```
And somehow test the logic against that?

Reply:
@ry   Now we get travis data in `main` function,  it will block the function. We can get this data in another asynchronous task.
for example:
in index.html
 `getTravisData()`
  `main();`

in app.js
`export async function getTravisData() {`
    `// when we got data success, we can store data to a json file or generate a chart directly.`
`}` 
In this case, they will not block each other.